zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCreationCreateProcessor.java

```diff
@@ -97,17 +102,30 @@ public ProcessInstanceCreationCreateProcessor(
   }
 
   @Override
-  public boolean onCommand(
+  public Either<Rejection, AuthorizationRequest<DeployedProcess>> getAuthorizationRequest(
       final TypedRecord<ProcessInstanceCreationRecord> command,
       final CommandControl<ProcessInstanceCreationRecord> controller) {
+    return getProcess(command.getValue())
+        .map(
+            process ->
+                new AuthorizationRequest<DeployedProcess>(
```

Comment:
💭 Adding the specific target resource as a generic to the authorization request might limit us very soon. Currently, we only work with what is called ["atomic checks" in C7](https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/main/java/org/camunda/bpm/engine/impl/db/PermissionCheckBuilder.java#L34). For those, a resource can be determined and is required so we can map it against the permissions defined internally. However, from experience, I'm relatively certain we'll want to allow "combined" permission checks soon as well, e.g. 
```
CREATE on any PROCESS_INSTANCE
OR
CREATE_INSTANCE on PROCESS_DEFINITION with bpmnProcessId=xyz
```
What is the resource generic of the `AuthorizationRequest` in those cases?
I see that it is appealing to fetch the target resource already in the `getAuthorizationRequest` and hand it over to the `onCommand` method but my gut feeling is that this will not hold very long. It looks like a different concern that we're patching onto the authorization request. At least that is what it looks like to me 🙂

That being said, it might still all work well and we're saving on fetching data multiple times. But it feels like a bandaid somehow 😄 And I believe the problem stems from the wrapper class, where we split up the processing logic now and thus need to fetch the target resource twice (once for getting the auth request, once for executing the command). That's why I suggested previously to keep those things within the original `onCommand` method, fetch the resource once, check the authorization using a behavior, and execute the logic if all is fine.

Reply:
I get what you're saying. When I [checked previously with PM](https://docs.google.com/document/d/1ikxw9ARCv-MBwASnj7WhW5uHCndCRk2B35xi-cB2N0M/edit?disco=AAABTovC3AY) it was mentioned that they don't see this as a requirements for 8.8 and don't have any feature requests that require this.

This isn't something that's unchangeable. We don't break backwards compatibility so if the request comes in the future we can solve it then 🤷 

I don't see it as a big problem personally. If you need multiple resources nothing is stopping you from adding a record `Resources(ResourceA resourceA, ResourceB resourceB)` and passing that object along.

Reply:
I agree, we can still change this later if we a) discover it's hindering us more than it helps and b) need to handle the target resource(s) differently.

The generic on the distributed command wrapper is still weird though, I have to say 😄 I am quite certain that this will lead to some confusion in the future as to why this is needed.

Reply:
Yeah I understand why you'd think that. If you think it's more clear we could also remove the generic and add a field to the classes to prevent fetching data multiple times? But my concern is that this will be even more confusing 😄

Reply:
Well, you could split up the commands further into `Resource getTargetResource()` (or whatever), `getAuthorizationRequest(Resource)`, and `onCommand(..., Resource)` (or whatever the method is depending on the type).

Reply:
Meh, yeah, right 😅

Reply:
Let's keep it as is for now since it's working. I'm still not entirely sure whether the wrapper is more harm or help. Let's see where this is going.